pkg/configurator/methods.go

```diff
@@ -182,3 +182,24 @@ func (c *Client) GetConfigResyncInterval() time.Duration {
 func (c *Client) GetProxyResources() corev1.ResourceRequirements {
 	return c.getMeshConfig().proxyResources
 }
+
+// GetInboundExternalAuthConfig returns the External Authentication configuration for incoming traffic, if any
+func (c *Client) GetInboundExternalAuthConfig() ExternAuthConfig {
+	extAuthConfig := ExternAuthConfig{}
+	inboundExtAuthzMeshConfig := c.getMeshConfig().inboundExternalAuth
+
+	extAuthConfig.Enable = inboundExtAuthzMeshConfig.Enable
+	extAuthConfig.Address = inboundExtAuthzMeshConfig.Address
+	extAuthConfig.Port = uint16(inboundExtAuthzMeshConfig.Port)
+	extAuthConfig.StatPrefix = inboundExtAuthzMeshConfig.StatPrefix
+	extAuthConfig.FailureModeAllow = inboundExtAuthzMeshConfig.FailureModeAllow
+
+	duration, err := time.ParseDuration(inboundExtAuthzMeshConfig.Timeout)
+	if err != nil {
+		log.Debug().Err(err).Msgf("ExternAuthzTimeout: Not a valid duration %s. defaulting to 1s.", duration)
+		duration = 1 * time.Second
```

Comment:
Why is a default being used here? Should this config be rejected instead so the user is aware?

Reply:
Not sure we can reject or consider the entire config nil if timeout fails to parse? what's your take on this?
IMO enforcing a correct parseable timeout string should be part of the Mesh CRD spec somehow. We can return empty config here if timeout fails to parse, but sounds like overkill to me.

Reply:
We will have a validator for this soon, so invalid values will be prevented.